src/two_captcha_client.js

```diff
@@ -0,0 +1,324 @@
+      id: captchaId,
+      action: 'get'
+    });
+
+    let decodedCaptcha = new Captcha();
+    decodedCaptcha.id = captchaId;
+    decodedCaptcha.apiResponse = res;
+
+    decodedCaptcha.text = res.split('|', 2)[1];
+
+    return decodedCaptcha;
+  }
+
+
+  /**
+   * Sends an image captcha and polls for its response
+   *
+   * @param  {Object} options          Parameters for the requests
+   * @param  {string} [options.base64] An already base64-coded image
+   * @param  {Buffer} [options.buffer] A buffer object of a binary image
+   * @param  {string} [options.path]   The path for a system-stored image
+   * @param  {string} [options.url]    Url for a web-located image
+   * @param  {string} [options.method] 2Captcha method of image sending. Can be either base64 or multipart
+   * @return {Promise<Captcha>}        Promise for a Captcha object
+   */
+  async decode(options = {}) {
+    let startedAt = Date.now();
+
+    if (typeof(this.key) !== 'string') this._throwError('2Captcha key must be a string')
+
+    let base64 = await this._loadCaptcha(options);
+
+    let decodedCaptcha = await this._upload({ ...options,
+      ...{
+        base64: base64
+      }
+    });
+
+    // Keep pooling untill the answer is ready
+    while (!decodedCaptcha.text) {
+      await this._sleep(this.polling);
+      if (Date.now() - startedAt > this.timeout) {
+        this._throwError('Captcha timeout');
+        return;
+      }
+      decodedCaptcha = await this.captcha(decodedCaptcha.id);
+    }
+
+    return decodedCaptcha;
+  }
+
+  /**
+   * Sends a ReCaptcha v2 and polls for its response
+   *
+   * @param  {Object} options           Parameters for the request
+   * @param  {string} options.googlekey The google key from the ReCaptcha
+   * @param  {string} options.pageurl   The URL where the ReCaptcha is
+   * @return {Promise<Captcha>}         Promise for a Captcha object
+   */
+  async decodeRecaptchaV2(options = {}) {
+    let startedAt = Date.now();
+
+    if (options.googlekey === '') this._throwError('Missing googlekey parameter');
+    if (options.pageurl === '') this._throwError('Missing pageurl parameter');
+
+    let upload_options = {
+      method: 'userrecaptcha',
+      googlekey: options.googlekey,
+      pageurl: options.pageurl
+    };
+
+    let decodedCaptcha = await this._upload(upload_options);
+
+    // Keep pooling untill the answer is ready
+    while (!decodedCaptcha.text) {
+      await this._sleep(Math.max(this.polling, 10)); // Sleep at least 10 seconds
+      if (Date.now() - startedAt > this.timeout) {
+        this._throwError('Captcha timeout');
+        return;
+      }
+      decodedCaptcha = await this.captcha(decodedCaptcha.id);
+    }
+
+    return decodedCaptcha;
+  }
+
+  /**
+   * @deprecated /load.php route is returning error 500
+   * Get current load from 2Captcha service
+   *
+   * @return {Promise<string>} Promise for an XML containing current load from
+   * 2Captcha service
+   */
+  async load() {
```

Comment:
Why adding if it is deprecated?

Reply:
Just keeping the same pattern from the Ruby TwoCaptcha gem. Also, I don't know if this route will work again some day.